vllm/worker/hpu_model_runner.py

```diff
@@ -382,6 +371,18 @@ def _set_block_scales(self, metadata, device):
         metadata = metadata._replace(block_scales=block_scales)
         return metadata
 
+    def _set_indices_and_offsets(self, metadata, block_size, is_prompt):
+        slot_mapping = metadata.slot_mapping.flatten()
+        indices = torch.div(slot_mapping, block_size, rounding_mode="floor")
+        if is_prompt:
+            indices = indices.unflatten(0, (-1, block_size))[:, 0]
+            offsets = None
+        else:
+            offsets = torch.fmod(slot_mapping, block_size)
+        metadata = metadata._replace(block_offsets=offsets)
+        metadata = metadata._replace(block_indices=indices)
```

Comment:
This can be made in a single call to _replace i.e. (block_offset=x, block_indices=y)

Reply:
Done